### PR TITLE
UIOR-405: Reset POForm after template change

### PR DESCRIPTION
**Pull request by Vladyslav-Velytskyi**
## Purpose

This fix helps form to updates when user change the template name
(before it , form does not update notes and checkboxes) 
Bug - https://issues.folio.org/browse/UIOR-405

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/PurchaseOrder/POForm.js [hunk 1]
```diff
@@ -178,11 +179,15 @@ class POForm extends Component {
   }
 
   onChangeTemplate = (e, value) => {
-    const { change, dispatch, parentResources, stripes } = this.props;
+    const { change, dispatch, reset, parentResources, stripes } = this.props;
     const templateValue = getOrderTemplateValue(parentResources, value);
     let form = get(stripes.store.getState(), 'form', {});
+    const { FormPO: { values: { poNumber } } } = form;
 
+    dispatch(reset('form'));
```
Comment (aliaksei-chumakou) [inline src/components/PurchaseOrder/POForm.js]: @Vladyslav-Velytskyi are you sure it should be `'form'`?

**Comment by aliaksei-chumakou** [comment]
@Vladyslav-Velytskyi how you were able to not use PR template? )

File: src/components/PurchaseOrder/POForm.js (hunk 1 above)
Comment (NikitaSedyx) [inline src/components/PurchaseOrder/POForm.js]: should form name be passed?

**Comment by NikitaSedyx** [comment]
@Vladyslav-Velytskyi please merge/rebase master in this branch

File: src/components/PurchaseOrder/POForm.js (hunk 1 above)
Comment (Vladyslav-Velytskyi) [reply src/components/PurchaseOrder/POForm.js]: @aliaksei-chumakou not really, I rechecked, `'form'` could be omit, thanks!

Comment (Vladyslav-Velytskyi) [reply src/components/PurchaseOrder/POForm.js]: No, it could work without it. Thanks!

**Pull request closed without merging**